### PR TITLE
[JENKINS-73156] handle svg cleanup of symbols via an xml document

### PR DESCRIPTION
**Pull request by mawinter69**
some attributes and elements where removed via regular expressions that removed more than what is necessary. Additionally `stroke:#000` was replaced with currentColor because there are some icons that still use that.
This had side-effects when symbols are used that are not just black/white, e.g. using svgs from jenkins artwork via customizable-header plugin. There are constructs like using class attributes and a style definition referring to the classes. At other places there where color definitions with `stroke:#000abc` which were incorrectly modified.

svg cleanup is now implemented with DOM manipulation which is more robust.
5 icons that still used `stroke:#000` are converted to use `stroke:currentColor`

opened https://github.com/jenkinsci/ionicons-api-plugin/pull/64 as followup to fix some occurrences of `stroke:#000`

See [JENKINS-73156](https://issues.jenkins.io/browse/JENKINS-73156).

### Testing done
Manual testing, that the changed icons still work properly with dark theme.
Used some of the SVGs from https://www.jenkins.io/artwork/ and provided them as symbol in customizable-header plugin and verified they are properly shown

### Proposed changelog entries

- Handle svg cleanup via an xml document to avoid broken symbols.

### Proposed upgrade guidelines

update ionicons-api plugin to at least version [74.v93d5eb_813d5f](https://github.com/jenkinsci/ionicons-api-plugin/releases/tag/74.v93d5eb_813d5f)

```[tasklist]
### Submitter checklist
- [x] The Jira issue, if it exists, is well-described.
- [x] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [x] There is automated testing or an explanation as to why this change has no tests.
- [x] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [x] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [x] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [x] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [x] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [x] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [x] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [x] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [x] Proper changelog labels are set so that the changelog can be generated automatically.
- [x] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [x] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

File: core/src/main/java/org/jenkins/ui/symbol/Symbol.java
Comment (mawinter69) [inline core/src/main/java/org/jenkins/ui/symbol/Symbol.java:113]: As before just emptying the title element. But why not just delete them completely?

**Review by timja — APPROVED** [review]

**Review by NotMyFault — APPROVED** [review]
/label ready-for-merge

---

This PR is now ready for merge. We will merge it after ~24 hours if there is no negative feedback.
Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process.
Thanks!

**Pull request merged**
